04_numeronym/srini/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	numeronyms := []string{}
+	for _, val := range vals {
+		numeronyms = append(numeronyms, generateNumeronym(val))
```

Comment:
This is fine, but if you really care about runtime you could optimise by preallocating the size of `numeronyms`:
```go
numeronyms := make([]string, len(vals))
for i, val := range vals {
		numeronyms[i] =  generateNumeronym(val)
}
```